idunn/blocks/happy_hour.py

```diff
@@ -13,22 +14,22 @@
 logger = logging.getLogger(__name__)
 
 
-class DaysType(types.Type):
-    dayofweek = validators.Integer(minimum=1, maximum=7)
-    local_date = validators.Date()
-    status = validators.String(enum=['yes', 'no'])
-    happy_hours = validators.Array(items=OpeningHoursType)
+class DaysType(BaseModel):
+    dayofweek: conint(ge=1, le=7)
+    local_date: date
+    status: constr(regex='(yes|no)')
+    happy_hours: List[OpeningHoursType]
 
 
 class HappyHourBlock(BaseBlock):
-    BLOCK_TYPE = 'happy_hours'
-    STATUSES = ['yes', 'no']
-
-    status = validators.String(enum=STATUSES)
-    next_transition_datetime = validators.String(allow_null=True)
-    seconds_before_next_transition = validators.Integer(allow_null=True)
-    raw = validators.String()
-    days = validators.Array(items=DaysType)
+    BLOCK_TYPE: ClassVar = 'happy_hours'
+    STATUSES: ClassVar = ['yes', 'no']
+
+    status: constr(regex='({})'.format('|'.join(STATUSES)))
```

Comment:
Maybe it would be cleaner to define an Enum ?  
https://pydantic-docs.helpmanual.io/usage/types/#enums-and-choices